driver-core/src/main/java/com/datastax/driver/core/querybuilder/Utils.java

```diff
@@ -32,7 +32,7 @@
 // Static utilities private to the query builder
 abstract class Utils {
 
-    private static final Pattern cnamePattern = Pattern.compile("\\w+(?:\\[.+\\])?");
+    private static final Pattern cnamePattern = Pattern.compile("\\w+(?:\\[.+\\]|\\.\\w+)?");
```

Comment:
I admit that this may not be the best route.  Also supporting this may break existing user code (i.e. if they have columns named `"myc.ol"`).